src/main/java/bobthebot/utils/Storage.java

```diff
@@ -0,0 +1,126 @@
+        this.dirPath = filePath.substring(0, filePath.lastIndexOf("/"));
+    }
+
+    /**
+     * Loads data from the specified file into an ArrayList, which can be used for the ToDo List object.
+     * If the file is not found, an empty ArrayList is returned.
+     *
+     * @return An ArrayList containing the tasks in the specified file, or an empty ArrayList if there is no
+     *     file or the file is empty.
+     */
+    public ArrayList<Task> load() {
+        ArrayList<Task> list = new ArrayList<>();
+        File dir = new File(this.dirPath);
+        if (!dir.exists()) {
+            dir.mkdir();
+        }
+
+        File file = new File(this.filePath);
+        if (dir.exists() && !file.exists()) {
+            try {
+                file.createNewFile();
+            } catch (IOException e) {
+                System.err.println("Not sure what an IOException is but it has occurred.");
+            }
+        }
+
+        try {
+            Scanner sc = new Scanner(file);
+
+            while (sc.hasNextLine()) {
+                String input = sc.nextLine();
+                System.out.println(input);
+                String[] parsedInput = input.split(" \\| ");
+                Task task;
+
+                switch (parsedInput[0]) {
+                case "T":
+                    task = new Todo(parsedInput[2]);
+                    break;
+                case "D":
+                    task = new Deadline(parsedInput[2], parsedInput[3]);
+                    break;
+                case "E":
+                    task = new Event(parsedInput[2], parsedInput[3]);
+                    break;
+                default:
+                    task = null;
+                    System.err.println("Error occurred during file loading. I do not process this task type.");
+                    break;
+                }
+
+                // checking if the task is done
+                if (Integer.parseInt(parsedInput[1]) == 1) {
+                    task.markDone();
+                }
+
+                list.add(task);
+            }
+
+            sc.close();
+
+        } catch (FileNotFoundException e) {
+            System.err.println(e);
+            return list;
+        }
+        return list;
+    }
+
+    /**
+     * Takes in an ArrayList of tasks and stores it into the specified file.
+     *
+     * @param list An ArrayList containing the Tasks in the ToDo list object.
+     */
+    public void store(ArrayList<Task> list) {
+        File dir = new File(this.dirPath);
+        if (!dir.exists()) {
+            dir.mkdir();
+        }
+
+        File file = new File(this.filePath);
+        if (dir.exists() && !file.exists()) {
+            try {
+                file.createNewFile();
+            } catch (IOException e) {
+                System.err.println("Not sure what an IOException is but it has occurred.");
+            }
+        }
+
+        try {
+            FileWriter fw = new FileWriter(this.filePath, false);

```

Comment:
```suggestion
            FileWriter fw = new FileWriter(this.filePath, isAppendMode);

```

I think it would be awesome to use an explicit variable like `isAppendMode` to avoid "magic" values.